src/FffGcodeWriter.cpp

```diff
@@ -3273,51 +3320,42 @@ bool FffGcodeWriter::processSkinPart(
 
     gcode_layer.mode_skip_agressive_merge_ = true;
 
-    processRoofing(storage, gcode_layer, mesh, extruder_nr, mesh_config, skin_part, added_something);
+    processRoofingFlooring(storage, gcode_layer, mesh, extruder_nr, "roofing", skin_part.roofing_fill, mesh_config.roofing_config, mesh.roofing_angles, added_something);
+    processRoofingFlooring(storage, gcode_layer, mesh, extruder_nr, "flooring", skin_part.flooring_fill, mesh_config.flooring_config, mesh.flooring_angles, added_something);
     processTopBottom(storage, gcode_layer, mesh, extruder_nr, mesh_config, skin_part, added_something);
 
     gcode_layer.mode_skip_agressive_merge_ = false;
     return added_something;
 }
 
-void FffGcodeWriter::processRoofing(
+void FffGcodeWriter::processRoofingFlooring(
     const SliceDataStorage& storage,
     LayerPlan& gcode_layer,
     const SliceMeshStorage& mesh,
     const size_t extruder_nr,
-    const MeshPathConfigs& mesh_config,
-    const SkinPart& skin_part,
+    const std::string& setting_prefix,
```

Comment:
Using string interpolation as a method of retrieving settings seems like a recipe for unmaintainable code.

I'd propose making it into a structure you can just pass to the function instead, like;

~~~
struct RoofFloorSettingNames
{
public:
std::string extruder_nr;
<...>
};
~~~

Fill those up for the two possibilities and you'd be able to use it like:

`const size_t skin_extruder_nr = mesh.settings.get<ExtruderTrain&>(rooffloor_settings.extruder_nr, setting_prefix)).extruder_nr_;`

Reply:
Good idea, I'll try it